client/src/ProtectedRoutes/ProtectedRoutes.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Navigate,
+} from 'react-router-dom';
+import { useAuth } from '../Context/AuthContext';
+
+export function ProtectedAdmin({ children }) {
+  const { user } = useAuth();
+  if (!user.user) {
+    return <Navigate to="/dashboard" replace />;
+  }
+  return children;
+}
+export function ProtectedUser({ children }) {
+  const { user } = useAuth();
+  if (user.user.role === 0) {
```

Comment:
and here you just want to check if the there is a user in the context
not the user role